### PR TITLE
chore: stop creating coderd for notification unit tests

### PR DESCRIPTION
**Pull request by spikecurtis**
A bunch of notification tests create a whole `coderd`, when all they use is the database and logger.  This makes the tests more expensive to run, and pollutes the test logs with a bunch of stuff that doesn't matter (e.g. tailnet).

### CODE REVIEW

**Comment by spikecurtis** [comment]
* **#15134** <a href="https://app.graphite.dev/github/pr/coder/coder/15134?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#15133** <a href="https://app.graphite.dev/github/pr/coder/coder/15133?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @spikecurtis and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: coderd/notifications/notifications_test.go [hunk 1]
```diff
@@ -460,7 +466,8 @@ func TestExpiredLeaseIsRequeued(t *testing.T) {
 
 	// nolint:gocritic // Unit test.
 	ctx := dbauthz.AsSystemRestricted(testutil.Context(t, testutil.WaitSuperLong))
-	_, _, api := coderdtest.NewWithAPI(t, nil)
+	store, _ := dbtestutil.NewDB(t)
+	logger := slogtest.Make(t, nil).Leveled(slog.LevelDebug)
```
Comment (dannykopping) [inline coderd/notifications/notifications_test.go:470]: Do we need to ignore errors on these?

**Review by dannykopping — APPROVED** [review]
Thanks for this! My unfamiliarity with our test utils was clearly evident here.

File: coderd/notifications/notifications_test.go (hunk 1 above)
Comment (spikecurtis) [reply coderd/notifications/notifications_test.go:470]: I don't believe so.  Does the manager or notifier drop error logs in this test case?

Note that `context.Canceled` is already ignored by default.

**Pull request merged**